packages/admin/admin/src/common/buttons/Button.tsx

```diff
@@ -0,0 +1,257 @@
+import { ThreeDotSaving } from "@comet/admin-icons";
+import {
+    Breakpoint,
+    Button as MuiButton,
+    ButtonProps as MuiButtonProps,
+    ComponentsOverrides,
+    css,
+    Theme,
+    useTheme,
+    useThemeProps,
+} from "@mui/material";
+import { ReactNode, useState } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { createComponentSlot } from "../../helpers/createComponentSlot";
+import { ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps";
+import { useWindowSize } from "../../helpers/useWindowSize";
+import { Tooltip } from "../Tooltip";
+
+type StateClassKey = "usingResponsiveBehavior" | "feedbackStateLoading" | "feedbackStateSuccess" | "feedbackStateError";
+type SlotClassKey = "root" | "mobileTooltip" | "successFeedback" | "errorFeedback";
+
+export type ButtonClassKey = StateClassKey | SlotClassKey;
+
+type ButtonThemeProps = ThemedComponentBaseProps<{
+    root: typeof MuiButton;
+    mobileTooltip: typeof Tooltip;
+    successFeedback: typeof Tooltip;
+    errorFeedback: typeof Tooltip;
+}>;
+
+type ResponsiveBehaviorSettings = {
+    breakpoint: Breakpoint;
+    mobileIcon: ReactNode;
+};
+
+type ResponsiveBehaviorOptions = Omit<Partial<ResponsiveBehaviorSettings>, "mobileIcon"> & {
+    mobileIcon?: "startIcon" | "endIcon" | ReactNode;
+};
+
+type FeedbackBehaviorOptions = {
+    state?: ButtonFeedbackState | null;
+    successMessage?: ReactNode;
+    errorMessage?: ReactNode;
+    successDuration?: number;
+    errorDuration?: number;
+};
+
+export type ButtonFeedbackState = "none" | "loading" | "success" | "error";
+
+type OwnerState = {
+    usingResponsiveBehavior: boolean;
+    feedbackState: ButtonFeedbackState;
+};
+
+type CustomButtonProps = {
+    responsiveBehavior?: boolean | ResponsiveBehaviorOptions;
+    feedbackBehavior?: boolean | FeedbackBehaviorOptions;
```

Comment:
@johnnyomair @thomasdax98 what do you think about this API, using the `responsiveBehavior` and `feedbackBehavior` props to activate/configure the buttons behavior? 

I think combining the `ToolbarActionButton` and `FeedbackButton` components into a single `Button` makes the most sense to allow a combination of those features and allow using the responsive behavior of `ToolbarActionButton` in buttons outside of Toolbars.